Macro error when compiling

**Describe the bug**

Hi, I would like to try your library but sadly tui 0.10 doesn't compile. Something with the crossterm backend and some macro apparently. Am I doing something wrong?

```
   Compiling tui v0.10.0
error[E0599]: no method named `flush` found for struct `std::string::String` in the current scope
  --> C:\Users\Jean\.cargo\registry\src\github.com-1ecc6299db9ec823\tui-0.10.0\src\backend\crossterm.rs:64:19
   |
64 |         map_error(queue!(string, MoveTo(0, 0)))?;
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `std::string::String`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
note: `backend::Backend` defines an item `flush`, perhaps you need to implement it
  --> C:\Users\Jean\.cargo\registry\src\github.com-1ecc6299db9ec823\tui-0.10.0\src\backend\mod.rs:29:1
   |
29 | pub trait Backend {
   | ^^^^^^^^^^^^^^^^^
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `flush` found for struct `std::string::String` in the current scope
  --> C:\Users\Jean\.cargo\registry\src\github.com-1ecc6299db9ec823\tui-0.10.0\src\backend\crossterm.rs:67:27
   |
67 |                 map_error(queue!(string, MoveTo(x, y)))?;
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `std::string::String`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
note: `backend::Backend` defines an item `flush`, perhaps you need to implement it
  --> C:\Users\Jean\.cargo\registry\src\github.com-1ecc6299db9ec823\tui-0.10.0\src\backend\mod.rs:29:1
   |
29 | pub trait Backend {
   | ^^^^^^^^^^^^^^^^^
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `flush` found for struct `std::string::String` in the current scope
  --> C:\Users\Jean\.cargo\registry\src\github.com-1ecc6299db9ec823\tui-0.10.0\src\backend\crossterm.rs:81:27
   |
81 |                 map_error(queue!(string, SetForegroundColor(color)))?;
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `std::string::String`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
note: `backend::Backend` defines an item `flush`, perhaps you need to implement it
  --> C:\Users\Jean\.cargo\registry\src\github.com-1ecc6299db9ec823\tui-0.10.0\src\backend\mod.rs:29:1
   |
29 | pub trait Backend {
   | ^^^^^^^^^^^^^^^^^
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `flush` found for struct `std::string::String` in the current scope
  --> C:\Users\Jean\.cargo\registry\src\github.com-1ecc6299db9ec823\tui-0.10.0\src\backend\crossterm.rs:86:27
   |
86 |                 map_error(queue!(string, SetBackgroundColor(color)))?;
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `std::string::String`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
note: `backend::Backend` defines an item `flush`, perhaps you need to implement it
  --> C:\Users\Jean\.cargo\registry\src\github.com-1ecc6299db9ec823\tui-0.10.0\src\backend\mod.rs:29:1
   |
29 | pub trait Backend {
   | ^^^^^^^^^^^^^^^^^
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `flush` found for type parameter `W` in the current scope
   --> C:\Users\Jean\.cargo\registry\src\github.com-1ecc6299db9ec823\tui-0.10.0\src\backend\crossterm.rs:242:23
    |
242 |             map_error(queue!(w, SetAttribute(CAttribute::NormalIntensity)))?;
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `W`
    |
    = help: items from traits can only be used if the type parameter is bounded by the trait
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: the following traits define an item `flush`, perhaps you need to restrict type parameter `W` with one of them:
    |
236 |     fn queue<W: backend::Backend>(&self, mut w: W) -> io::Result<()>
    |              ^^^^^^^^^^^^^^^^^^^
236 |     fn queue<W: std::io::Write>(&self, mut w: W) -> io::Result<()>
    |              ^^^^^^^^^^^^^^^^^

error[E0599]: no method named `flush` found for type parameter `W` in the current scope
   --> C:\Users\Jean\.cargo\registry\src\github.com-1ecc6299db9ec823\tui-0.10.0\src\backend\crossterm.rs:245:23
    |
245 |             map_error(queue!(w, SetAttribute(CAttribute::NoUnderline)))?;
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `W`
    |
    = help: items from traits can only be used if the type parameter is bounded by the trait
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: the following traits define an item `flush`, perhaps you need to restrict type parameter `W` with one of them:
    |
236 |     fn queue<W: backend::Backend>(&self, mut w: W) -> io::Result<()>
    |              ^^^^^^^^^^^^^^^^^^^
236 |     fn queue<W: std::io::Write>(&self, mut w: W) -> io::Result<()>
    |              ^^^^^^^^^^^^^^^^^

error[E0599]: no method named `flush` found for type parameter `W` in the current scope
   --> C:\Users\Jean\.cargo\registry\src\github.com-1ecc6299db9ec823\tui-0.10.0\src\backend\crossterm.rs:250:23
    |
250 |             map_error(queue!(w, SetAttribute(CAttribute::Bold)))?;
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `W`
    |
    = help: items from traits can only be used if the type parameter is bounded by the trait
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: the following traits define an item `flush`, perhaps you need to restrict type parameter `W` with one of them:
    |
236 |     fn queue<W: backend::Backend>(&self, mut w: W) -> io::Result<()>
    |              ^^^^^^^^^^^^^^^^^^^
236 |     fn queue<W: std::io::Write>(&self, mut w: W) -> io::Result<()>
    |              ^^^^^^^^^^^^^^^^^

error[E0599]: no method named `flush` found for type parameter `W` in the current scope
   --> C:\Users\Jean\.cargo\registry\src\github.com-1ecc6299db9ec823\tui-0.10.0\src\backend\crossterm.rs:253:23
    |
253 |             map_error(queue!(w, SetAttribute(CAttribute::Underlined)))?;
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `W`
    |
    = help: items from traits can only be used if the type parameter is bounded by the trait
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: the following traits define an item `flush`, perhaps you need to restrict type parameter `W` with one of them:
    |
236 |     fn queue<W: backend::Backend>(&self, mut w: W) -> io::Result<()>
    |              ^^^^^^^^^^^^^^^^^^^
236 |     fn queue<W: std::io::Write>(&self, mut w: W) -> io::Result<()>
    |              ^^^^^^^^^^^^^^^^^

error: aborting due to 8 previous errors
```

**To Reproduce**

I added Tui with crossterm as dependencies
```toml
[dependencies]
tui = { version = "0.10", features = ["crossterm"], default-features = false }
```
```sh
cargo build
```

Interestingly, 0.8 compiles and 0.9 doesn't.

```toml
[dependencies]
tui = { version = "0.8", features = ["crossterm"], default-features = false }
```

Demos don't work either, I get the same error.

```sh
cargo run --example crossterm_demo --no-default-features --features="crossterm" --release -- --tick-rate 200
```

**Expected behavior**

Tui to compile. 

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Terminal Emulator: N/A
 - Font: N/A
 - Crate version: from 0.9 to 0.10
 - Backend: crossterm

**Additional context**

Toolchain: 
stable-x86_64-pc-windows-msvc (default)
rustc 1.46.0 (04488afe3 2020-08-24)

--- a/src/backend/crossterm.rs
+++ b/src/backend/crossterm.rs
@@ -165,7 +165,6 @@ struct ModifierDiff {
     pub to: Modifier,
 }
 
-#[cfg(unix)]
 impl ModifierDiff {
     fn queue<W>(&self, mut w: W) -> io::Result<()>
     where
@@ -224,31 +223,6 @@ impl ModifierDiff {
             map_error(queue!(w, SetAttribute(CAttribute::RapidBlink)))?;
         }
 
-        Ok(())
-    }
-}
-
-#[cfg(windows)]
-impl ModifierDiff {
-    fn queue<W>(&self, mut w: W) -> io::Result<()>
-    where
-        W: fmt::Write,
-    {
-        let removed = self.from - self.to;
-        if removed.contains(Modifier::BOLD) {
-            map_error(queue!(w, SetAttribute(CAttribute::NormalIntensity)))?;
-        }
-        if removed.contains(Modifier::UNDERLINED) {
-            map_error(queue!(w, SetAttribute(CAttribute::NoUnderline)))?;
-        }
-
-        let added = self.to - self.from;
-        if added.contains(Modifier::BOLD) {
-            map_error(queue!(w, SetAttribute(CAttribute::Bold)))?;
-        }
-        if added.contains(Modifier::UNDERLINED) {
-            map_error(queue!(w, SetAttribute(CAttribute::Underlined)))?;
-        }
         Ok(())
     }
 }

--- a/src/backend/crossterm.rs
+++ b/src/backend/crossterm.rs
@@ -13,10 +13,7 @@ use crossterm::{
     },
     terminal::{self, Clear, ClearType},
 };
-use std::{
-    fmt,
-    io::{self, Write},
-};
+use std::io::{self, Write};
 
 pub struct CrosstermBackend<W: Write> {
     buffer: W,
@@ -52,9 +49,6 @@ where
     where
         I: Iterator<Item = (u16, u16, &'a Cell)>,
     {
-        use fmt::Write;
-
-        let mut string = String::with_capacity(content.size_hint().0 * 3);
         let mut fg = Color::Reset;
         let mut bg = Color::Reset;
         let mut modifier = Modifier::empty();
@@ -62,7 +56,7 @@ where
         for (x, y, cell) in content {
             // Move the cursor if the previous location was not (x - 1, y)
             if !matches!(last_pos, Some(p) if x == p.0 + 1 && y == p.1) {
-                map_error(queue!(string, MoveTo(x, y)))?;
+                map_error(queue!(self.buffer, MoveTo(x, y)))?;
             }
             last_pos = Some((x, y));
             if cell.modifier != modifier {
@@ -70,26 +64,25 @@ where
                     from: modifier,
                     to: cell.modifier,
                 };
-                diff.queue(&mut string)?;
+                diff.queue(&mut self.buffer)?;
                 modifier = cell.modifier;
             }
             if cell.fg != fg {
                 let color = CColor::from(cell.fg);
-                map_error(queue!(string, SetForegroundColor(color)))?;
+                map_error(queue!(self.buffer, SetForegroundColor(color)))?;
                 fg = cell.fg;
             }
             if cell.bg != bg {
                 let color = CColor::from(cell.bg);
-                map_error(queue!(string, SetBackgroundColor(color)))?;
+                map_error(queue!(self.buffer, SetBackgroundColor(color)))?;
                 bg = cell.bg;
             }
 
-            string.push_str(&cell.symbol);
+            map_error(queue!(self.buffer, Print(&cell.symbol)))?;
         }
 
         map_error(queue!(
             self.buffer,
-            Print(string),
             SetForegroundColor(CColor::Reset),
             SetBackgroundColor(CColor::Reset),
             SetAttribute(CAttribute::Reset)
@@ -168,7 +161,7 @@ struct ModifierDiff {
 impl ModifierDiff {
     fn queue<W>(&self, mut w: W) -> io::Result<()>
     where
-        W: fmt::Write,
+        W: io::Write,
     {
         //use crossterm::Attribute;
         let removed = self.from - self.to;

